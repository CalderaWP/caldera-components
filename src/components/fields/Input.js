import React from 'react';
import PropTypes from 'prop-types';

export const Input = (props) => {
	return (
		<input
			type={props.inputType ? props.inputType : 'text'}
			id={props.id}
			className={props.fieldClassName}
			aria-describedby={props.ariaDescribedbyAttr}
			required={props.isRequired}
		/>
	)

};

Input.propTypes = {
	id: PropTypes.string.isRequired,
	fieldClassName: PropTypes.string.isRequired,
	inputType: PropTypes.string,
	ariaDescribedbyAttr: PropTypes.string,
	isRequired: PropTypes.bool
};

Input.defaultProps = {
	ariaDescribedbyAttr: '',
	isRequired: false
};