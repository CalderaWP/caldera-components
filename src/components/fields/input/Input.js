import React from 'react';
import {fieldPropTypes} from '../propTypes';
import classNames from 'classnames';
import {RenderGroup} from "../../RenderGroup";
/**
 * Input component for any HTML5 input type
 *
 * @param {Object} props
 * @returns {*}
 * @constructor
 */
export const Input = (props) => {
	/**
	 * Dispatches value of input when it changes
	 * @param event
	 * @return {*}
	 */
	function changeHandler(event){
		return props.onValueChange(event.target.value);
	}

	/**
	 * Dispatches value when checkbox is checked
	 * @return {*}
	 */
	function checkboxChangeHandler() {
		return props.onValueChange(! props.value);
	}


	/**
	 * Get the className prop for inputs
	 *
	 * @return {String}
	 */
	function inputClassName() {
		return classNames(
			props.fieldClassName,
			RenderGroup.classNames.input
		)
	}

	if( 'checkbox' === props.inputType ){
		return (
			<input
				type={'checkbox'}
				id={props.id}
				className={inputClassName()}
				aria-describedby={props.ariaDescribedbyAttr}
				required={props.isRequired}
				onChange={checkboxChangeHandler}
				defaultChecked={props.value}
			/>
		);
	}

	return (
		<input
			type={props.inputType}
			id={props.id}
			className={inputClassName()}
			aria-describedby={props.ariaDescribedbyAttr}
			required={props.isRequired}
			onChange={changeHandler}
			value={props.value}
		/>
	);

};

/**
 * Prop definitions for Input components
 */
Input.propTypes = fieldPropTypes;

/**
 * Default props for Input Component
 *
 * @type {{ariaDescribedbyAttr: string, isRequired: boolean, inputType: string}}
 */
Input.defaultProps = {
	ariaDescribedbyAttr: '',
	isRequired: false,
	inputType: 'text'
};