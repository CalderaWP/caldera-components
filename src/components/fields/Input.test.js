import renderer from 'react-test-renderer';
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Input} from "./Input";
Enzyme.configure({ adapter: new Adapter() });

describe( 'Input component', () => {
	describe( 'Input component props', () => {
		it( 'Passes its props', () => {
			const component = renderer.create(
				<Input
					id={'bags'}
					fieldClassName={'foo'}
				/>
			);
			expect( component.toJSON() ).toMatchSnapshot();
		})
	});

	describe( 'Input component rendering', () => {
		it( 'Has the correct id attribute', () => {
			const wrapper = shallow(
				<Input
					id={'bags'}
					fieldClassName={'foo'}
				/>
			);
			expect( wrapper.find( '#bags').length).toBe(1);
		});

		it( 'has the correct class attribute', () => {
			const wrapper = shallow(
				<Input
					id={'bags'}
					fieldClassName={'foo'}
				/>
			);
			expect( wrapper.find( '.foo').length).toBe(1);
		});
	});
});
