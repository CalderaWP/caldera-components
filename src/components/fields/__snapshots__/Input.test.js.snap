// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Input component props Passes its props 1`] = `
<input
  aria-describedby=""
  className="foo"
  id="bags"
  required={false}
  type="text"
/>
`;
